lib/board.rb

```diff
@@ -0,0 +1,52 @@
+class Board
+    
+    attr_accessor :cells
+    
+    def initialize
+        reset!
+    end
+    
+    def reset!
+        @cells = Array.new(9, " ")
+    end
+    
+    def display
+     puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+     puts "-----------"
+     puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+     puts "-----------"
+     puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    end
+    
+    def position(number)
+        cell_number = (number.to_i)-1
+        return "#{@cells[cell_number]}"
+    end
+    
+    def full?
+        cells.all?{|token| token == "X" || token == "O"}
+    end
+    
+    def turn_count
+        cells.count{|token| token == "X" || token == "O"}
+    end
+    
+    def taken?(number)
+        cell_number = (number.to_i)-1
+        cells[cell_number] == 'X' || cells[cell_number] == 'O' ? true : false
+        
+    end
+    
+    def valid_move?(number)
+        (1..9).any? {|item| item == number.to_i} && !taken?(number) 
```

Comment:
You can simplify the integer range by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F). This keeps you from having to iterate over the entire range.
`number.to_i.between?(1,9)`